plugins/modules/net_tools/nmcli.py

```diff
@@ -1030,9 +1068,14 @@ def connection_update(self, nmcli_command):
         if self.type == "wifi":
             cmd.append('ssid')
             cmd.append(self.ssid)
+            if self.wifi:
+                for name, value in self.wifi.items():
+                    if name == 'ssid':
+                        continue
+                    cmd += ['802-11-wireless.%s' % name, value]
             if self.wifi_sec:
                 for name, value in self.wifi_sec.items():
-                    cmd += ['wifi-sec.%s' % name, value]
+                    cmd += ['802-11-wireless-security.%s' % name, value]
```

Comment:
I don't see any mention of this in #2220, but writing secrets to command line is a security hole. If possible I believe it would be best to add the `-a` or `--ask` flag to prompt for the password to be passed via `stdin`. `nmcli` also can be directed to a file containing the password which is preferable to command line, but still risky. @felixfontein please chime in if you think this is not relevant for this use case.

Snippet from the `nmcli` man pages:
```
 passwd-file
               some networks may require credentials during activation. You can give these credentials using this
               option. Each line of the file should contain one password in the form:

                   setting_name.property_name:the password

               For example, for WPA Wi-Fi with PSK, the line would be

                   802-11-wireless-security.psk:secret12345

               For 802.1X password, the line would be

                   802-1x.password:my 1X password

               nmcli also accepts wifi-sec and wifi strings instead of 802-11-wireless-security. When NetworkManager
               requires a password and it is not given, nmcli will ask for it when run with --ask. If --ask was not
               passed, NetworkManager can ask another secret agent that may be running (typically a GUI secret agent,
               such as nm-applet or gnome-shell).
```

Reply:
Hello @Ajpantuso, I agree with your statement, however this pull request may not be the proper place for that fix. The only reason there are `wifi_sec`-related modifications is for adding the compare functionality. Perhaps I should remove those modifications and add a new pull request?

Reply:
Yeah, I guess the path then would be:
- Limit this PR to the new wifi options
- Follow-up with a bugfix PR with the password related changes which backports to `stable-3`
- Introduce `wifi-sec` mods in a subsequent feature PR

This module tends to have multiple open PR's so not sure if there's any other conflicting ones out there however.